napari/_qt/tests/test_qt_viewer.py

```diff
@@ -316,3 +316,83 @@ def test_qt_viewer_data_integrity(qtbot, dtype):
     viewer.dims.ndisplay = 2
     datamean = viewer.layers[0].data.mean()
     assert datamean == imean
+
+
+@pytest.fixture()
+def view(qtbot):
+
+    viewer = ViewerModel()
+    view = QtViewer(viewer)
+    qtbot.addWidget(view)
+
+    np.random.seed(0)
+    data = np.random.random((10, 10, 15))
+    viewer.add_image(data)
+
+    def increment(e):
+        view.dims.counter += 1
+        # if we don't "enable play" again, view.dims won't request a new frame
+        view.dims._play_ready = True
+
+    view.dims.dims.events.axis.connect(increment)
+    view.dims.counter = 0
```

Comment:
Could you set this before the definition of `increment`? Having it under it breaks my internal parser. =P